_includes/assets/js/indicatorModelHelpers.js

```diff
@@ -0,0 +1,539 @@
+(function() {
+  return {
+    UNIT_COLUMN: 'Units',
+    GEOCODE_COLUMN: 'GeoCode',
+    YEAR_COLUMN: 'Year',
+    VALUE_COLUMN: 'Value',
+    convertJsonFormat: function(data) {
+      var keys = Object.keys(data);
+      if (!keys.length) {
+        return [];
+      }
+
+      return data[keys[0]].map(function(item, index) {
+        return _.object(keys, keys.map(function(key) {
+          return data[key][index];
+        }));
+      });
+    },
+    extractUnique: function(prop, data) {
+      return _.chain(data).pluck(prop).uniq().filter(function(f) { return f; }).sortBy(function(year) {
+        return year;
+      }).value();
+    },
+    getColumnsFromData: function(data) {
+      return Object.keys(data[0]);
+    },
+    dataHasColumn: function(column, data) {
+      return this.getColumnsFromData(data).includes(column);
+    },
+    dataHasUnits: function(data) {
+      return this.dataHasColumn(this.UNIT_COLUMN, data);
+    },
+    dataHasGeoCodes: function(data) {
+      return this.dataHasColumn(this.GEOCODE_COLUMN, data);
+    },
+    getFirstUnitInData: function(data) {
+      return data.find(function(row) {
+        return row[this.UNIT_COLUMN];
+      }, this)[this.UNIT_COLUMN];
+    },
+    getDataByUnit: function(data, unit) {
+      if (!unit) {
```

Comment:
Is `!unit` supposed to check if the `unit` parameter is empty (ie. not provided)? If yes, then the function shouldn't return the `data` if `unit` is not provided, it should throw an error instead since `unit` is essential for the function